File: app/src/main/java/com/example/toothfairy/DateChangedReceiver.java
package com.example.toothfairy;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import androidx.lifecycle.ViewModelProvider;

import com.example.toothfairy.viewModel.MainViewModel;

public class DateChangedReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        // TODO: This method is called when the BroadcastReceiver is receiving

        switch (intent.getAction()){
            case Intent.ACTION_TIME_CHANGED:
                Toast.makeText(context, "날짜 바뀜", Toast.LENGTH_SHORT).show();

                // 당일 착용 시간 갱신
                
                // 당일 착용 시간으로 WearingStats 갱신
                /* MainViewModel의 updatePatientStats() 호출 */

                // DailyWearingTime 테이블 갱신
                break;
            case Intent.ACTION_TIME_TICK:
                Toast.makeText(context, "1분 지남", Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/com/example/toothfairy/DateChangedReceiver.kt
package com.example.toothfairy

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.example.toothfairy.model.repository.WearingInfoRepository
import com.example.toothfairy.util.NotifyManager
import java.lang.Exception

/** 날짜 변경을 감지하는 리시버 클래스 */
class DateChangedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        // TODO: This method is called when the BroadcastReceiver is receiving
        try{
            WearingInfoRepository.saveDailyWearingTime()
        }catch (e:Exception){
            Log.e("에러", e.printStackTrace().toString())

            NotifyManager.sendNotification(
                context,
                NotifyManager.WEAR_RECOMMEND_NOTIFY_ID,
                "에러 내용",
                e.message.toString()
            )
        }

    }
}